### PR TITLE
Go fix dataflows new

### LINKED ISSUE #6: Golang query tests are currently failing
```
$ codeql test run ./go/test/
Executing 4 tests in 4 directories.
Extracting test database in /go/test/query-tests/crypto/MsgNotHashedBeforeSigVerfication.
Compiling queries in /go/test/query-tests/crypto/MsgNotHashedBeforeSigVerfication.
Executing tests in /go/test/query-tests/crypto/MsgNotHashedBeforeSigVerfication.
--- expected
+++ actual
@@ -1,6 +1,11 @@
+WARNING: Module PathGraph has been deprecated and may be removed in future (MsgNotHashedBeforeSigVerfication.ql:14,8-27)
+WARNING: Type Configuration has been deprecated and may be removed in future (MsgNotHashedBeforeSigVerfication.ql:93,33-57)
+WARNING: Type Configuration has been deprecated and may be removed in future (MsgNotHashedBeforeSigVerfication.ql:103,50-73)
+WARNING: Type PathNode has been deprecated and may be removed in future (MsgNotHashedBeforeSigVerfication.ql:146,49-67)
+WARNING: Type PathNode has been deprecated and may be removed in future (MsgNotHashedBeforeSigVerfication.ql:146,76-94)
 edges
-| MsgNotHashedBeforeSigVerfication.go:78:33:78:46 | "test message" | MsgNotHashedBeforeSigVerfication.go:78:26:78:47 | type conversion |
-| MsgNotHashedBeforeSigVerfication.go:86:31:86:44 | "test message" | MsgNotHashedBeforeSigVerfication.go:86:24:86:45 | type conversion |
+| MsgNotHashedBeforeSigVerfication.go:78:33:78:46 | "test message" | MsgNotHashedBeforeSigVerfication.go:78:26:78:47 | type conversion | provenance |  |
+| MsgNotHashedBeforeSigVerfication.go:86:31:86:44 | "test message" | MsgNotHashedBeforeSigVerfication.go:86:24:86:45 | type conversion | provenance |  |
 nodes
 | MsgNotHashedBeforeSigVerfication.go:78:26:78:47 | type conversion | semmle.label | type conversion |
 | MsgNotHashedBeforeSigVerfication.go:78:33:78:46 | "test message" | semmle.label | "test message" |
[1/4 comp 384ms eval 1.2s] FAILED(RESULT) /go/test/query-tests/crypto/MsgNotHashedBeforeSigVerfication/MsgNotHashedBeforeSigVerfication.qlref
Extracting test database in /go/test/query-tests/security/FilePermsFlaws.
Compiling queries in /go/test/query-tests/security/FilePermsFlaws.
Executing tests in /go/test/query-tests/security/FilePermsFlaws.
[2/4 comp 85ms eval 40ms] PASSED /go/test/query-tests/security/FilePermsFlaws/FilePermsFlaws.qlref
Extracting test database in /go/test/query-tests/security/TrimMisuse.
Compiling queries in /go/test/query-tests/security/TrimMisuse.
Executing tests in /go/test/query-tests/security/TrimMisuse.
--- expected
+++ actual
@@ -1,3 +1,4 @@
+WARNING: Type Configuration has been deprecated and may be removed in future (TrimMisuse.ql:19,26-49)
 | TrimMisuse.go:14:14:14:45 | call to TrimRight | TrimRight will consider the second argument ($@) as a list of elements, and not as indivisible string. | TrimMisuse.go:14:40:14:44 | "BBA" | BBA |
 | TrimMisuse.go:15:14:15:51 | call to TrimRight | TrimRight will consider the second argument ($@) as a list of elements, and not as indivisible string. | TrimMisuse.go:15:46:15:50 | "BBA" | BBA |
 | TrimMisuse.go:21:14:21:42 | call to TrimRight | TrimRight will consider the second argument ($@) as a list of elements, and not as indivisible string. | TrimMisuse.go:20:7:20:13 | "/test" | /test |
[3/4 comp 134ms eval 440ms] FAILED(RESULT) /go/test/query-tests/security/TrimMisuse/TrimMisuse.qlref
Extracting test database in /go/test/query-tests/security/MissingMinVersionTLS.
Compiling queries in /go/test/query-tests/security/MissingMinVersionTLS.
Executing tests in /go/test/query-tests/security/MissingMinVersionTLS.
--- expected
+++ actual
@@ -1,3 +1,5 @@
+WARNING: Type Configuration has been deprecated and may be removed in future (MissingMinVersionTLS.ql:18,36-64)
+WARNING: Type Configuration has been deprecated and may be removed in future (MissingMinVersionTLS.ql:45,33-61)
 | MissingMinVersionTLS.go:25:14:25:25 | struct literal | TLS.Config.MinVersion is never set for variable $@  | MissingMinVersionTLS.go:25:3:25:8 | config | config |
 | MissingMinVersionTLS.go:35:14:37:3 | struct literal | TLS.Config.MinVersion is never set for variable $@  | MissingMinVersionTLS.go:35:3:35:8 | config | config |
 | MissingMinVersionTLS.go:50:13:50:24 | struct literal | TLS.Config.MinVersion is never set for variable $@  | MissingMinVersionTLS.go:50:3:50:8 | config | config |
[4/4 comp 128ms eval 705ms] FAILED(RESULT) /go/test/query-tests/security/MissingMinVersionTLS/MissingMinVersionTLS.qlref
Completed in 12.7s (extract 8.4s comp 731ms eval 2.4s).
1 tests passed; 3 tests failed:
  FAILED: /go/test/query-tests/crypto/MsgNotHashedBeforeSigVerfication/MsgNotHashedBeforeSigVerfication.qlref
  FAILED: /go/test/query-tests/security/MissingMinVersionTLS/MissingMinVersionTLS.qlref
  FAILED: /go/test/query-tests/security/TrimMisuse/TrimMisuse.qlref
```

I suspect it may have something to do with this: https://github.blog/changelog/2023-08-14-new-dataflow-api-for-writing-custom-codeql-queries/. We should update our queries to avoid the deprecated functionality.

### PR DESCRIPTION
**Pull request by GrosQuildu**
Fixes https://github.com/trailofbits/codeql-queries/issues/6

### CODE REVIEW

File: go/test/query-tests/crypto/MsgNotHashedBeforeSigVerfication/MsgNotHashedBeforeSigVerfication.expected [hunk 1]
```diff
@@ -1,6 +1,6 @@
 edges
-| MsgNotHashedBeforeSigVerfication.go:78:33:78:46 | "test message" | MsgNotHashedBeforeSigVerfication.go:78:26:78:47 | type conversion |
-| MsgNotHashedBeforeSigVerfication.go:86:31:86:44 | "test message" | MsgNotHashedBeforeSigVerfication.go:86:24:86:45 | type conversion |
+| MsgNotHashedBeforeSigVerfication.go:78:33:78:46 | "test message" | MsgNotHashedBeforeSigVerfication.go:78:26:78:47 | type conversion | provenance |  |
```
Comment (mschwager) [inline go/test/query-tests/crypto/MsgNotHashedBeforeSigVerfication/MsgNotHashedBeforeSigVerfication.expected:2]: Any idea why `provenance` was added here?

**Review by mschwager — APPROVED** [review]
Nice work! Thanks for fixing up the tests

File: go/test/query-tests/crypto/MsgNotHashedBeforeSigVerfication/MsgNotHashedBeforeSigVerfication.expected (hunk 1 above)
Comment (GrosQuildu) [reply go/test/query-tests/crypto/MsgNotHashedBeforeSigVerfication/MsgNotHashedBeforeSigVerfication.expected:2]: Could be something with https://codeql.github.com/docs/codeql-overview/codeql-changelog/codeql-cli-2.10.0/#c
but I don't understand codeql's testing format for path queries

Comment (mschwager) [reply go/test/query-tests/crypto/MsgNotHashedBeforeSigVerfication/MsgNotHashedBeforeSigVerfication.expected:2]: Sounds good 👍 no need to block on this comment, I was just curious.

**Pull request merged**